packages/engine/Source/Scene/Model/ModelInstance.js

```diff
@@ -0,0 +1,251 @@
+   * @readonly
+   */
+  get relativeTransform() {
+    return this._relativeTransform;
+  }
+
+  /**
+   * The Pick Id of the instance.
+   * @type {string|undefined}
+   * @readonly
+   */
+  get pickId() {
+    return this._pickId;
+  }
+
+  _updateTransform(transform) {
+    // Get center from the transform
+    this._center = Matrix4.getTranslation(transform, this._center);
+
+    // Get relative transform from the center
+    const translationRotationScale = scratchTranslationRotationScale;
+    translationRotationScale.translation = Cartesian3.ZERO;
+    translationRotationScale.scale = Matrix4.getScale(
+      transform,
+      translationRotationScale.scale,
+    );
+
+    const rotation = Matrix4.getRotation(transform, scratchRotation);
+    translationRotationScale.rotation = Quaternion.fromRotationMatrix(
+      rotation,
+      translationRotationScale.rotation,
+    );
+
+    this._relativeTransform = Matrix4.fromTranslationRotationScale(
+      translationRotationScale,
+    );
+  }
+
+  /**
+   * Compute the matrix to transform the instance from the root model
+   * node to world space, taking into accont a top-level modelMatrix
+   * applied to each instance in the collection, and the axis correction
+   * from glTF's up-axis to Cesium's worldspace up-axis.
+   * @param {Matrix4} modelMatrix The model's modelMatrix property
+   * @param {Matrix4} rootTransform The root node transform to model-space
+   * @param {Matrix4} [result] If provided, the instance in which to store the result.
+   * @returns {Matrix4} The computed matrix
+   * @private
+   */
+  computeModelMatrix(modelMatrix, rootTransform, result) {
+    result = result ?? new Matrix4();
+
+    const instanceTransform = this.transform;
+    let transform = Matrix4.multiplyTransformation(
+      instanceTransform,
+      rootTransform,
+      result,
+    );
+    transform = Matrix4.multiplyTransformation(modelMatrix, transform, result);
+    return transform;
+  }
+
+  /**
+   * Get a bounding sphere that encapsulates a model instance.
+   * @param {Model} model The model being instanced.
+   * @param {BoundingSphere} [result] If provided, the instance in which to store the result.
+   * @returns {BoundingSphere} The model instance bounding sphere.
+   * @example
+   * // Fly to a specific model instance
+   * const model = await Cesium.Model.fromGltfAsync({
+   *   url: "../../SampleData/models/GroundVehicle/GroundVehicle.glb",
+   *   minimumPixelSize: 64,
+   * });
+   * viewer.scene.primitives.add(model);
+   *
+   * const modelInstance = model.instances.add(instanceModelMatrix);
+   *
+   * const boundingSphere = modelInstance.getBoundingSphere(model);
+   * viewer.camera.flyToBoundingSphere(boundingSphere);
+   */
+  getBoundingSphere(model, result) {
+    const modelMatrix = model.modelMatrix;
+    const sceneGraph = model.sceneGraph;
+    const instanceBoundingSpheres = [];
+
+    // TODO: runtime nodes should probably have a internal public accessor
```

Comment:
this does not seem problematic to add.

does this also indicate that ModelSceneGraph should be a public class?